openquake/hazardlib/tests/gsim/mgmpe/generic_gmpe_avgsa_test.py

```diff
@@ -0,0 +1,101 @@
+# The Hazard Library
+# Copyright (C) 2012-2019 GEM Foundation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from openquake.hazardlib import gsim, imt, const
+import numpy as np
+import unittest
+
+class GenericGmpeAvgSATestCase(unittest.TestCase):
+
+    _ = gsim.get_available_gsims()
+
+    def test_calculation_Akkar(self):
+        """
+        """
+
+        DATA_FILE = 'Data/GENERIC_GMPE_AVGSA_MEAN_STD_TOTAL_AKKAR.csv'
+
+        # Initialise meta-GMPE
+        mgmpe = gsim.mgmpe.generic_gmpe_avgsa.GenericGmpeAvgSA(
+                    gmpe_name='BooreAtkinson2008', 
+                    avg_periods="0.05,0.15,1.0,2.0,4.0",
+                    corr_func='akkar')
+
+        sctx = gsim.base.SitesContext()
+        rctx = gsim.base.RuptureContext()
+        dctx = gsim.base.DistancesContext()
+
+        P = imt.SA(1.0)
```

Comment:
This is confusing as you are setting an IMT that is not used in practice. It would be cleaner to introduce an ad-hoc intensity measure type e.g. AveragedSa or something similar.

Reply:
Fixed